### ISSUE TITLE
200 Response Code When Exception is Thrown During Bootstrap

### ISSUE THREAD

**Issue by Nuranto** [magento/magento2#30847]
### Preconditions (*)

1. Version 2.4.1
2. PHP 7.4

### Steps to reproduce (*)

1. Change your DB host to something wrong.

### Expected result (*)

1. An error message is displayed and a 500 response code is returned.

### Actual result (*)

1. An error message is displayed and the http response code is 200 OK.

Very annoying for probes who think the pod is OK & Running...

### Additiona; info from Engcom
full env.php (Removed crypt key) : 
```
<?php

return [
    'backend' => [
        'frontName' => 'admin',
    ],
    'queue' => [
        'consumers_wait_for_messages' => 1,
    ],
    'crypt' => [
        'key' => '****',
    ],
    'db' => [
        'table_prefix' => '',
        'connection' => [
            'default' => [
                'host' => 'nonexisting',
                'dbname' => 'whatever',
                'username' => 'root',
                'password' => 'root',
                'model' => 'mysql4',
                'engine' => 'innodb',
                'initStatements' => 'SET NAMES utf8;',
                'active' => '1',
                'driver_options' => [
                    1014 => false,
                ],
            ],
        ],
    ],
    'resource' => [
        'default_setup' => [
            'connection' => 'default',
        ],
    ],
    'x-frame-options' => 'SAMEORIGIN',
    'MAGE_MODE' => 'production',
    'session' => [
        'save' => 'files',
    ],
    'cache' => [
        'frontend' => [
            'default' => [
                'id_prefix' => '707_',
            ],
            'page_cache' => [
                'id_prefix' => '707_',
            ],
        ],
        'allow_parallel_generation' => false,
    ],
    'lock' => [
        'provider' => 'db',
        'config' => [
            'prefix' => '',
        ],
    ],
    'cache_types' => [
        'config' => 0,
        'layout' => 0,
        'block_html' => 0,
        'collections' => 0,
        'reflection' => 0,
        'db_ddl' => 0,
        'compiled_config' => 1,
        'eav' => 0,
        'customer_notification' => 0,
        'config_integration' => 0,
        'config_integration_api' => 0,
        'full_page' => 0,
        'config_webservice' => 0,
        'translate' => 0,
        'vertex' => 0,
    ],
    'downloadable_domains' => [
    ],
    'install' => [
        'date' => 'Fri, 09 Oct 2020 12:13:31 +0000',
    ],
];

```

I havn't any extra module enabled.

![Capture d’écran 2020-11-09 à 15 10 32](https://user-images.githubusercontent.com/1633463/98551684-e1e4a500-229d-11eb-808f-84eabd16faa7.png)

**Comment by engcom-Bravo** [comment]
Hello @Nuranto 

Thank you for your report

We have rechecked the reported issue with the provided steps and we are **not** able to reproduce the _200 Response Code_

We have changed the `db host` in _env.php_ file to wrong. 
![env_host](https://user-images.githubusercontent.com/51680745/98540764-16089780-2297-11eb-9f15-ff3fd26fb32c.png)
Set Magento into Production Mode `php bin/magento dep:mo:set production`

What we get on Admin Page and on Storefront, is a _There has been an error processing your request_ error and a :x:_GET  500 (Internal Server Error)_

We are not able to reproduce the _http response code is 200_

Can you please provide more info about the issue if possible with screenshots?

Thak you

**Comment by Nuranto** [comment]
Thank you @engcom-Bravo.

I get a different message from you : `An error has happened during application run. See exception log for details.`

My full env.php (I just removed crypt key) : 
```
<?php

return [
    'backend' => [
        'frontName' => 'admin',
    ],
    'queue' => [
        'consumers_wait_for_messages' => 1,
    ],
    'crypt' => [
        'key' => '****',
    ],
    'db' => [
        'table_prefix' => '',
        'connection' => [
            'default' => [
                'host' => 'nonexisting',
                'dbname' => 'whatever',
                'username' => 'root',
                'password' => 'root',
                'model' => 'mysql4',
                'engine' => 'innodb',
                'initStatements' => 'SET NAMES utf8;',
                'active' => '1',
                'driver_options' => [
                    1014 => false,
                ],
            ],
        ],
    ],
    'resource' => [
        'default_setup' => [
            'connection' => 'default',
        ],
    ],
    'x-frame-options' => 'SAMEORIGIN',
    'MAGE_MODE' => 'production',
    'session' => [
        'save' => 'files',
    ],
    'cache' => [
        'frontend' => [
            'default' => [
                'id_prefix' => '707_',
            ],
            'page_cache' => [
                'id_prefix' => '707_',
            ],
        ],
        'allow_parallel_generation' => false,
    ],
    'lock' => [
        'provider' => 'db',
        'config' => [
            'prefix' => '',
        ],
    ],
    'cache_types' => [
        'config' => 0,
        'layout' => 0,
        'block_html' => 0,
        'collections' => 0,
        'reflection' => 0,
        'db_ddl' => 0,
        'compiled_config' => 1,
        'eav' => 0,
        'customer_notification' => 0,
        'config_integration' => 0,
        'config_integration_api' => 0,
        'full_page' => 0,
        'config_webservice' => 0,
        'translate' => 0,
        'vertex' => 0,
    ],
    'downloadable_domains' => [
    ],
    'install' => [
        'date' => 'Fri, 09 Oct 2020 12:13:31 +0000',
    ],
];

```

I havn't any extra module enabled.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Bravo 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-38992` were created

**Issue Available**: @engcom-Bravo, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by devamitbera** [comment]
@magento I am working on this

**Comment by devamitbera** [comment]
### Update
The issue happens when config cache is disabled  `'config' => 0,`

**Comment by ihor-sviziev** [comment]
This issue was fixed in https://github.com/magento/magento2/commit/fd8749f6ecea40dcd759b1bf76cb68695737fadf. I'm closing this issue as fixed.

**Comment by magento-engcom-team** [comment]
Hi @Nuranto, @devamitbera. 

Thank you for your report and collaboration!

The related internal Jira ticket `MC-38992` was closed as `Fixed`. 

The fix will be available with the upcoming `2.4.3` release.
